lib/velum/salt.rb

```diff
@@ -34,6 +35,27 @@ def self.update_status(targets: "*", cached: false)
       [needed["return"], failed["return"]]
     end
 
+    # Trigger salt-cloud to construct a cluster
+    def self.build_cloud_cluster(count)
+      threads = (1..count).collect do
+        instance_name = "caasp-node-" + SecureRandom.hex(4)
+        Thread.new do
+          Thread.current[:response] = perform_request(
+            endpoint: "/",
+            method:   "post",
+            data:     {
+              client: "local_async",
+              tgt:    "admin",
+              fun:    "cloud.profile",
+              arg:    ["cluster_node", instance_name]
+            }
+          )
+        end
+      end
+      threads.each(&:join)
+      threads.collect { |t| t[:response] }
```

Comment:
Is it really needed to thread this calls? After all we are joining them at the very end, so the request will not see any speedup. Since it's an async client, why not just loop-call `perform_request` without spawning threads?

Reply:
Or even better, if the looping happened in `cloud.profile` we could just do one single call with `arg: ["cluster_node", list_of_machines]`

Reply:
I can't find any example of `cloud.profile` being called with a list of machine names, where did you find that usage?

https://docs.saltstack.com/en/latest/topics/cloud/salt.html#cloud-profile

Reply:
You are right, didn't know where `cloud.profile` was coming from. Thanks for removing the thread spawning.